### DISCUSSION TITLE
Install brew for a chroot user but permission issues when installing chromium

### DISCUSSION THREAD

**Discussion by adrien-robert** [Homebrew/discussions#1897] [Getting started]
Hello,

I use GridPane service on one of my server and I need to install brew to use chromium headless inside a WP website.
I downloaded brew:
`mkdir homebrew && curl -L https://github.com/Homebrew/brew/tarball/master | tar xz --strip 1 -C homebrew`

Then I gave ownership of the folder and everything inside to my chroot user:
`chown -R mychrootuser:mychrootuser homebrew`

But now when I tried to install chromium with brew as my chrootuser I have multiple permissions issues:

root@myserver:/home/mychrootuser# `sudo -u mychrootuser homebrew/bin/brew install --cask chromium`

> mkdir: cannot create directory ‘/root’: Permission denied
> ==> Downloading https://ghcr.io/v2/homebrew/portable-ruby/portable-ruby/blobs/sha256:97e639a64dcec285392b53ad804b5334c324f1d2a8bdc2b5087b8bf8051e332f
> #=#=-  #       #
> Warning: /root/.cache/Homebrew/portable-ruby-2.6.3_2.x86_64_linux.bottle.tar.gz
> Warning: .incomplete: Permission denied
> 
> curl: (23) Failed writing body (0 != 1369)
> ==> Downloading https://github.com/Homebrew/homebrew-portable-ruby/releases/download/2.6.3_2/portable-ruby-2.6.3_2.x86_64_linux.bottle.tar.gz
> #=#=-  #       #
> Warning: /root/.cache/Homebrew/portable-ruby-2.6.3_2.x86_64_linux.bottle.tar.gz
> Warning: .incomplete: Permission denied
> 
> curl: (23) Failed writing body (0 != 934)
> Error: Failed to download https://ghcr.io/v2/homebrew/portable-ruby/portable-ruby/blobs/sha256:97e639a64dcec285392b53ad804b5334c324f1d2a8bdc2b5087b8bf8051e332f and https://github.com/Homebrew/homebrew-portable-ruby/rele
> 
> Do not file an issue on GitHub about this; you will need to figure out for
> yourself what issue with your internet connection restricts your access to
> GitHub (used for Homebrew updates and binary packages).
> Error: Failed to install Homebrew Portable Ruby and cannot find another Ruby 2.6.3!
> If there's no Homebrew Portable Ruby available for your processor:
> - install Ruby 2.6.3 with your system package manager (or rbenv/ruby-build)
> - make it first in your PATH
> - try again

I also tried to run it as root so I don't have issues but it's not allowed:

root@myserver:/home/mychrootuser# `homebrew/bin/brew install --cask chromium`

> Error: Running Homebrew as root is extremely dangerous and no longer supported.
> As Homebrew does not drop privileges on installation you would be giving all
> build scripts full access to your system.

Any help on this please?

**Comment by adrien-robert** [comment]
Alright, this is what I did so far:

- Create a .cache folder on the same path of my homebrew folder (with root acces)
- Giving permissions / ownership to mychrootuser
- `sudo -i -u mychrootuser`
- `homebrew/bin/brew install --cask chromium`

Then it worked and there was download.
But at the end I had this "Error: Cask 'chromium' is unavailable: No Cask with this name exists.":
![image](https://user-images.githubusercontent.com/4320990/127195329-f7f5e1d7-e5e9-4d3d-a4fc-6cb52fb7abbb.png)

**Comment by gromgit** [comment]
> root@myserver:/home/mychrootuser# `sudo -u mychrootuser homebrew/bin/brew install --cask chromium`

Here's the problem: This runs `brew` as user `mychrootuser`, but _with the existing (`root`) environment_, which includes `HOME` pointing to `/root` instead of `/home/mychrootuser`.

You've already found the `-i` flag to `sudo`, but note that it can be used with the command "inline" as well:
```
sudo -i -u mychrootuser homebrew/bin/brew install --cask chromium
```
That said, there's no getting around `Error: Cask 'chromium' is unavailable: No Cask with this name exists.`, because casks are a macOS-only concept.

**Reply by adrien-robert** [reply]
Alright, so how can I install chromium with brew on Ubuntu 18.04?

**Reply by SMillerDev** [reply]
You can't

**Reply by adrien-robert** [reply]
Alright so I'll try with puppeteer then for my headless browser
Thanks for answering!
